### ISSUE TITLE
User is kicked when attempting to use 'Explorer's compass' in a server

### ISSUE THREAD

**Issue by Lord-Deimos** [MattCzyr/ExplorersCompass#5]
Steps to reproduce:
1) Launch modded title
2) Join a modded server
3) Acquire Explorer's compass
4) Right-click to use Explorer's compass.
5) Observe as the user is kicked out of the server. 

Description: 
After the user acquires the explorer's compass and uses it with right-click while in online play, it causes the user to be kicked from the server.
Note: Using this while in single-player works as intended.

Result (3/3): The user is kicked from an online server while using Explorer's compass. Expected Result: Explorer's compass functions as intended without booting the user.

Expected Result: Explorer's compass functions as intended without kicking the user.

**Comment by MattCzyr** [comment]
What version of the mod are you using and for what mod loader? For Fabric, this should have been fixed in version 2.0.2.

**Comment by Lord-Deimos** [comment]
I am using the Fabric 2.0.2 version. The modloader, hm, I am unsure of what that is exactly, I am thinking that's what Curseforge is, right? Or would it be something like Fabric-loader-0.11.6-1.17.1? I'll just give these details

I am using Curseforge. The modpack I am using is Another Quality Modpack 2. I hope this helps.

**Comment by MattCzyr** [comment]
After getting kicked from the server, can you grab your client's latest.log and post it here? You can find it at .minecraft/logs/latest.log.

**Comment by Lord-Deimos** [comment]
Sorry about that. I would have done it yesterday but have been busy. Here's the log!
[latest.log](https://github.com/MattCzyr/ExplorersCompass/files/7190783/latest.log)

**Comment by Lord-Deimos** [comment]
Any idea what's going on?

**Comment by MattCzyr** [comment]
Try this beta file and let me know if the issue persists:

https://www.dropbox.com/s/arz3fr893sx0pnk/ExplorersCompass-1.17.1-2.0.3-fabric-beta.jar?dl=0

**Comment by jbreckel** [comment]
not OP but I encpuntered the same error.
Tried with the beta and I still get the same error. (beta on client and server)

Error:
```
[09:02:10] [Netty Client IO #2/ERROR]: Encountered exception while handling in channel with name "explorerscompass:sync"
net.minecraft.class_151: Non [a-z0-9/._-] character in path of location: minecraft:
	at net.minecraft.class_2960.<init>(class_2960.java:41) ~[intermediary-1.17.1.jar:?]
	at net.minecraft.class_2960.<init>(class_2960.java:46) ~[intermediary-1.17.1.jar:?]
	at net.minecraft.class_2540.method_10810(class_2540.java:527) ~[intermediary-1.17.1.jar:?]
	at com.chaosthedude.explorerscompass.network.SyncPacket.apply(SyncPacket.java:44) ~[ExplorersCompass-1.17.1-2.0.3-fabric-beta.jar:?]
```

**Comment by Ro-Wind** [comment]
The same issue is occurring on 1.18.1, 1.17 versions are completely fine.

**Comment by Vexalus** [comment]
Still seems to be occurring.

**Comment by MattCzyr** [comment]
Just released version 1.18.1-2.0.3, which should fix this. Feel free to reopen if it persists.
